internal/plugintest/working_dir.go

```diff
@@ -444,3 +526,38 @@ func (wd *WorkingDir) Schemas(ctx context.Context) (*tfjson.ProviderSchemas, err
 
 	return providerSchemas, err
 }
+
+func (wd *WorkingDir) Query(ctx context.Context) ([]string, error) {
+	logging.HelperResourceTrace(ctx, "Calling Terraform CLI providers query command")
+
+	var reattachStr string
+	if wd.reattachInfo != nil {
+		reattachJson, err := json.Marshal(wd.reattachInfo)
+		if err != nil {
+			return nil, fmt.Errorf("error marshaling reattach info: %w", err)
+		}
+		reattachStr = string(reattachJson)
+	}
+
+	buffer := &bytes.Buffer{}
+	errBuffer := &bytes.Buffer{}
+	cmd := exec.CommandContext(ctx, wd.terraformExec, "query", "-json")
+	cmd.Dir = wd.baseDir
+	cmd.Stdout = buffer
+	cmd.Stderr = errBuffer
+	cmd.Env = os.Environ()
+	fmt.Println("Reattach info: ", reattachStr)
+	cmd.Env = append(cmd.Env, "TF_REATTACH_PROVIDERS="+reattachStr)
+
+	err := cmd.Run()
+	if err != nil {
+		output := buffer.String() + "\n\n" + errBuffer.String()
+		fmt.Println(output)
+		return nil, fmt.Errorf("error running terraform query command: %w", err)
+	}
+
+	logging.HelperResourceTrace(ctx, "Called Terraform CLI providers query command")
+
+	output := buffer.String()
+	return []string{output}, nil
```

Comment:
Someday, https://github.com/hashicorp/terraform-json may parse this output. Until then, we may want to json-decode it.